sfklString.cpp

```diff
@@ -18,14 +18,68 @@
 // along with sfArkLib.  If not, see <http://www.gnu.org/licenses/>.
 
 #include <string.h>
+#include <stdio.h>
+#include "wcc.h"
 
-const char *GetFileExt(const char *FileName)
+char *GetFileExt(char *FileName)
 {
   // return pointer to (final) file extension including the dot e.g. '.txt'
-  const char *p;
+  // returns NULL if file is not found
+  char *p;
 
-  for(p = FileName + strlen(FileName); p > FileName &&  *p != '.'; p--) {}
-  if (*p == '.')  p = FileName + strlen(FileName);
-	return p;
+  for(p = FileName + strlen(FileName); p > FileName; p--)
+  {
+    if (*p == '.')
+    {
+      return p;
+    }
+    else if (*p == '/' || *p == '\\')
+    {
+      return NULL;
+    }
+  }
+  
+  return NULL;
 }
 
+char *StrncpyEnsureNul(char *destination, const char *source, int num)
+{
+  if (num == 0 || !destination || !source)
+  {
+    return destination;

```

Comment:
perhaps for safety `destination[0] = 0` (unless `!destination`)?

Reply:
Good idea. Safer the better.